js/global_variables/cell_names_array.js

```diff
@@ -1,5 +1,29 @@
 export const set_cell_names_array = (cats, cell_arrow_table) => {
+
+    // Extract the array of names (strings)
     cats.cell_names_array = cell_arrow_table.getChild("name").toArray();
+
+    // Create a set of unique names
+    const uniqueNames = [...new Set(cats.cell_names_array)];
```

Comment:
I'm not sure this is necessary and will this change the order of the names and their associated indexes? I think we can just loop through the array using reduce without first getting the unique names.

Reply:
Checked and everything looks good